### PR TITLE
[Elastic Agent] Fixed status reporting

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Fixed some issues discovered on testing.
First one is not to use overwrites when in standalone mode second one is that status code was always 200. (written after body) 

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by kvch — APPROVED** [review]
LGTM, I haven't tested it yet.

**Comment by kvch** [comment]
Also, test failure is going to be fixed by: https://github.com/elastic/beats/pull/25587

**Comment by kvch** [comment]
Tested, works well.

File: x-pack/elastic-agent/pkg/core/monitoring/server/handler.go [hunk 1]
```diff
@@ -27,15 +27,15 @@ type apiHandler struct {
...
-
 		switch e := err.(type) {
 		case apiError:
 			w.WriteHeader(e.Status())
 		default:
 			w.WriteHeader(http.StatusInternalServerError)
 
 		}
+
+		writeResponse(w, unexpectedErrorWithReason(err.Error()))
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/core/monitoring/server/handler.go:38]: Possible to add a unit test for this code path? Having a unit test would have been an easy way to catch this.

**Pull request merged**